src/ecc.rs

```diff
@@ -283,15 +283,36 @@ impl Ecc {
             }
 
             let response = EccResponse::from_bytes(&buf[..])?;
-            if idle {
-                self.transport.send_idle();
-            }
+            
             match response {
-                EccResponse::Data(bytes) => return Ok(bytes),
+                EccResponse::Data(bytes) => {
+                    if idle {
+                        self.transport.send_idle();
+                    }
+                    return Ok(bytes)
+                },
                 EccResponse::Error(err) if err.is_recoverable() && retry < retries => continue,
-                EccResponse::Error(err) => return Err(Error::ecc(err)),
+                EccResponse::Error(err) => {
+                    self.error_mitigation();
+                    return Err(Error::ecc(err));
+                }
             }
         }
+        self.error_mitigation();
```

Comment:
With this addition the sleep at line 278 can also be removed since that will fall through to here.